nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -65,7 +65,7 @@ def from_path(cls: Type["Entry"], path: Path) -> "Entry":
         # Matching nixos*-generation-$number*.conf
         rex_generation = re.compile(r"^nixos.*-generation-([0-9]+).*\.conf$")
         # Matching nixos*-generation-$number-specialisation-$specialisation_name*.conf
-        rex_specialisation = re.compile(r"^nixos.*-generation-([0-9]+)-specialisation-([a-zA-Z0-9]+).*\.conf$")
+        rex_specialisation = re.compile(r"^nixos.*-generation-([0-9]+)-specialisation-(.*)\.conf$")
```

Comment:
What would speak against expanding the allowed characters?

Reply:
The option is defined here https://github.com/NixOS/nixpkgs/blob/a58bc8ad779655e790115244571758e8de055e3d/nixos/modules/system/activation/specialisation.nix#L42

attrsOf doesn't seem to validate the name https://github.com/NixOS/nixpkgs/blob/0cb2fd7c59fed0cd82ef858cbcbdb552b9a33465/lib/types.nix#L577-L594

so anything that is allowed in the key of an attr.